src/Microsoft.Fx.Portability/Analyzer/RequestAnalyzer.cs

```diff
@@ -32,16 +32,22 @@ public AnalyzeResponse AnalyzeRequest(AnalyzeRequest request, string submissionI
                 .OrderBy(x => x.FullName, StringComparer.OrdinalIgnoreCase)
                 .ToList();
 
+            List<NuGetPackageInfo> nugetPackagesForUserAssemblies = new List<NuGetPackageInfo>();
+            var assembliesToRemove = _analysisEngine.ComputeAssembliesToRemove(request.UserAssemblies, targets, out nugetPackagesForUserAssemblies);
```

Comment:
You are calling `assemblieToRemove.Contains` in a loop, but `assembliesToRemove` is a list - if you switch stuff to use IEnumerable, then here put it into a `HashSet` so there is better lookup

Reply:
Changed into a HashSet